tool/src/test/resources/org/datacommons/tool/genmcf/successtmcf/output/summary_report.html

```diff
@@ -817,6 +823,25 @@ <h2>
                     </tr>
+                    <tr>
+                      <td><a href="#places--CA-SK--SV1" name="places--CA-SK--SV1">SV1</a></td>
+                      <td>1</td>
+                      <td>2019</td>
+                      <td>65</td>
```

Comment:
this is an error. the header and source rows in the CSV is:
```
Year,Place_Id,Place_Name,Place_Type,LatLong,SV1,SV2
2019,CA-SK,Saskatchewan,Province,[LatLong 43N 116W],"[5 6]",2
```

that is, the value `"[5 6]"` is being rendered as a `65`.

TODO(snny) investigate this

Reply:
I have investigated this. The tool is understanding `"[5 6]"` as a complex value meaning of "`5` of the unit `6`".

A few issues here;
1. Is the test case intentional? Before this PR allowed NaN SVObs values, there was no corresponding output in `summary_report.html` for this row, so it is hard to understand the desired behavior.
  - There is, however, a counter in `report.json` for `6` not being a valid unit. [(link)](https://github.com/datacommonsorg/import/blob/1f3b72bac5ddd86d57d87c64f8790f4f4617e376/tool/src/test/resources/org/datacommons/tool/genmcf/successtmcf/output/report.json#L279-L287)
2. `ComplexValueParser.java` attempts to understand if the value or the unit is given first in a complex value of the format `[value1 value2]` by checking if the first digit of the first field is a digit [(link)](https://github.com/datacommonsorg/import/blob/1f3b72bac5ddd86d57d87c64f8790f4f4617e376/util/src/main/java/org/datacommons/util/ComplexValueParser.java#L118), however, this fails because this test case seemingly has a numeric unit.
3. Even with such a parsing, the resulting TypedValue is `type: RESOLVED_REF` and `value: 65`. The `type` seems correct, but the `value` is unexpected; shouldn't the unit come later?

Two further thoughts about better supporting NaN values in StatChecker;
1. We should display the "type" of a series (facet) to make it clear how the values were parsed.
2. How should we display complex values in `summary_report.html`? This might not be as important to cover if we implement (1) and do not expect real-world imports to have "numeric" units (even if they do; that error will be surfaced as a failed existence check for the unit)

@pradh could you please review and let me know your thoughts?

Reply:
> We should display the "type" of a series (facet) to make it clear how the values were parsed.

https://github.com/datacommonsorg/import/pull/174 is an implementation of this. It is in draft status until this PR is merged to master.

Reply:
Good catch!

Yeah, the complex-value parsing could use some improvement.

So, in the default flag mode, where we don't allow string values for obs, does it error out?

I worry that the value-type is really implementation detail and adding a column for it is not very useful in *most* cases (e.g., unless its a NUMBER its a failure).  Perhaps we add a column only when something is a non-NUMBER? That is, if the measurement-result flag is enabled.

Reply:
> So, in the default flag mode, where we don't allow string values for obs, does it error out?

TODO(snny) to look into it

> Perhaps we add a column only when something is a non-NUMBER? That is, if the measurement-result flag is enabled.

Good idea. I will update #174 accordingly (TODO(snny)) and we can use that to decide if we want to keep or discard that change.

Reply:
> So, in the default flag mode, where we don't allow string values for obs, does it error out?

Yes, however, the report.json counter value is still displayed as `65`, which is confusing. See relevant counter below.

```
    "level": "LEVEL_ERROR",
    "location": {
      "file": "SuccessTmcf.csv",
      "lineNumber": "13"
    },
    "userMessage": "Found a non-numeric value for the value field of StatVarObservation but allowNanSVObs was false :: value: '65', property: 'value', node: 'E:SVTest->E0'",
    "counterKey": "Sanity_SVObs_Value_NotANumber"
```

Reply:
That's ok. As long as it gets parsed as a string.  If there's some way to report the original value that might be nice.

Reply:
I am working on reporting the type as it is parsed, since it is directly relevant to the error. Original value would likely touch  deep in the parsing logic.

Reply:
See: 870477d368074f84653e35e838c78fcdb3d8ad0a